gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/FieldValue.java

```diff
@@ -0,0 +1,236 @@
+     */
+    REPEATED,
+
+    /**
+     * A {@code FieldValue} for a field of type {@link Field.Type#record(Field...)}.
+     */
+    RECORD
+  }
+
+  FieldValue(Kind kind, Object value) {
+    this.kind = kind;
+    this.value = value;
+  }
+
+  /**
+   * Returns the kind of this Field Value.
+   *
+   * @return {@link Kind#PRIMITIVE} if the value is of primitive type ({@link Field.Type#bool()},
+   *     {@link Field.Type#string()}, {@link Field.Type#floatingPoint()},
+   *     {@link Field.Type#integer()}, {@link Field.Type#timestamp()}) or is {@code null}. Returns
+   *     {@link Kind#REPEATED} if the corresponding field has ({@link Field.Mode#REPEATED}) mode.
+   *     Returns {@link Kind#RECORD} if the corresponding field has
+   *     {@link Field.Type#record(Field...)} type.
+   */
+  public Kind kind() {
+    return kind;
+  }
+
+  /**
+   * Return this field's value as an {@link Object}.
+   */
+  public Object value() {
+    return value;
+  }
+
+  /**
+   * Return this field's value as a {@link String}. This method should only be used if the
+   * corresponding field has primitive type ({@link Field.Type#bool()}, {@link Field.Type#string()},
+   * {@link Field.Type#floatingPoint()}, {@link Field.Type#integer()},
+   * {@link Field.Type#timestamp()}).
+   *
+   * @throws ClassCastException if the field has not primitive type
+   */
+  @SuppressWarnings("unchecked")
+  public String stringValue() {
+    return (String) value;
+  }
+
+  /**
+   * Returns this field's value as a {@link Long}. This method should only be used if the
+   * corresponding field has {@link Field.Type#integer()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws NumberFormatException if the field's value could not be converted to {@link Integer}
+   */
+  @SuppressWarnings("unchecked")
+  public long longValue() {
+    return Long.valueOf(stringValue());
+  }
+
+  /**
+   * Returns this field's value as a {@link Double}. This method should only be used if the
+   * corresponding field has {@link Field.Type#floatingPoint()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws NumberFormatException if the field's value could not be converted to {@link Double}
+   */
+  @SuppressWarnings("unchecked")
+  public double doubleValue() {
+    return Double.valueOf(stringValue());
+  }
+
+  /**
+   * Returns this field's value as a {@link Boolean}. This method should only be used if the
+   * corresponding field has {@link Field.Type#bool()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws IllegalStateException if the field's value could not be converted to {@link Boolean}
+   */
+  @SuppressWarnings("unchecked")
+  public boolean booleanValue() {
+    String stringValue = stringValue();
+    checkState(stringValue.equalsIgnoreCase("true") || stringValue.equalsIgnoreCase("false"),
```

Comment:
I assume because we are using the json format it is always going to be "true" or "false" and this class will never be used for a CSV format which also supports 1 or 0?